CMakeLists.txt

```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.1)
 # project settings
 
 set(target SnailSnap)
-project(${target})
+project(${target} CXX)
```

Comment:
Why do we need this? CXX is already the default value for the project language. Is there any benefit in explicitly specifying it?

Reply:
CMake does only generate and read variables in the cache relevant to the languages used by the project. Trying to specify options regarding an unused language results in an error and can thusly prevent mistakes in certain situations, e.g.:
```
$ cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ..
-- The CXX compiler identification is Clang 6.0.0
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER
<snip>
```